### ISSUE TITLE
Question about the support of BlockDiagonalMask with Batched inputs 

### ISSUE THREAD

**Issue by ma787639046** [facebookresearch/xformers#760]
# ❓ Questions and Help
Thanks for your excellent work. 

I want to implement batched training with `xformers.ops.fmha.attn_bias.BlockDiagonalMask`, referring to https://facebookresearch.github.io/xformers/components/ops.html#xformers.ops.fmha.attn_bias.BlockDiagonalCausalMask. It seems that the optimized mask `BlockDiagonalMask` only support with `batch_size=1`, which can only concatenate multiple sequences from the `seq_len` dimension, but not the `batch_size` dimension. I wonder if I can create a `BlockDiagonalMask` with `batch_size>1`, which is useful for continuous concatenation of input sequences in the batched pre-training.

Thanks!
